src/Sparrow/Logging/LoggingSource.cs

```diff
@@ -141,7 +142,7 @@ public void SetupLogMode(LogMode logMode, string path)
                     // have to do this on a separate thread
                     Task.Run(() =>
                     {
-                        _keepLogging = false;
+                        _keepLogging.LowerOrDie();
```

Comment:
You must not do this here. This will kill the entire process if `OrDie` is called, since no one is catching the exception

Reply:
Replaced all .RaiseOrDie or .LowerOrDie for their not dying alternatives (which was the original behavior in all places anyways). Functions are left there so that people may start implementing with stronger invariants in mind.